code/modules/mob/living/carbon/xenomorph/egg.dm

```diff
@@ -0,0 +1,197 @@
+	///What type of hugger are produced here
+	var/hugger_type = /obj/item/clothing/mask/facehugger
+
+/obj/effect/alien/egg/hugger/update_icon_state()
+	. = ..()
+	overlays.Cut()
+	if(on_fire)
+		overlays += "alienegg_fire"
+	if(hivenumber != XENO_HIVE_NORMAL && GLOB.hive_datums[hivenumber])
+		var/datum/hive_status/hive = GLOB.hive_datums[hivenumber]
+		color = hive.color
+		return
+	color = null
+
+/obj/effect/alien/egg/hugger/burst(kill)
+	. = ..()
+	if(kill)
+		hugger_type = null
+		playsound(loc, "sound/effects/alien_egg_burst.ogg", 30)
+		flick("egg exploding", src)
+		return
+	advance_maturity(2)
+	playsound(src.loc, "sound/effects/alien_egg_move.ogg", 25)
+	flick("egg opening", src)
+	var/hugger = new hugger_type()
+	hugger_type = null
+	addtimer(CALLBACK(hugger, /atom/movable.proc/forceMove, loc), 1 SECONDS)
+
+/obj/effect/alien/egg/hugger/attack_alien(mob/living/carbon/xenomorph/xenomorph, damage_amount = xenomorph.xeno_caste.melee_damage, damage_type = BRUTE, damage_flag = "", effects = TRUE, armor_penetration = 0, isrightclick = FALSE)
+	if(xenomorph.status_flags & INCORPOREAL)
+		return FALSE
+
+	if(!istype(xenomorph))
+		return attack_hand(xenomorph)
+
+	if(!issamexenohive(xenomorph))
+		xenomorph.do_attack_animation(src, ATTACK_EFFECT_SMASH)
+		xenomorph.visible_message("<span class='xenowarning'>[xenomorph] crushes \the [src].","<span class='xenowarning'>We crush \the [src].")
+		burst(FALSE)
+		return
+
+	switch(maturity_stage)
+		if(0)
+			to_chat(xenomorph, span_xenowarning("The child is not developed yet."))
+		if(1)
+			to_chat(xenomorph, span_xenonotice("We retrieve the child."))
+			burst()
+		if(2)
+			xenomorph.visible_message(span_xenonotice("\The [xenomorph] clears the hatched egg."), \
+			span_xenonotice("We clear the hatched egg."))
+			playsound(loc, "alien_resin_break", 25)
+			qdel(src)
+
+/obj/effect/alien/egg/hugger/attackby(obj/item/I, mob/user, params)
+	. = ..()
+
+	if(!istype(I, /obj/item/clothing/mask/facehugger))
+		return FALSE
+	return insert_new_hugger(I, user)
+
+///Try to insert a new hugger into the egg
+/obj/effect/alien/egg/hugger/proc/insert_new_hugger(obj/item/clothing/mask/facehugger/facehugger, mob/user)
+	if(facehugger.stat == DEAD)
+		if(user)
+			to_chat(user, span_xenowarning("This child is dead."))
+		return FALSE
+
+	if(maturity_stage != stage_ready_to_burst + 1)
+		if(user)
+			to_chat(user, span_xenowarning("This egg is not usable"))
+		return FALSE
+
+	if(hugger_type)
+		if(user)
+			to_chat(user, span_xenowarning("This one is occupied with a child."))
+		return FALSE
+	if(user)
+		visible_message(span_xenowarning("[user] slides [facehugger] back into [src]."),span_xenonotice("You place the child into [src]."))
+	hugger_type = facehugger.type
+	qdel(facehugger)
+	advance_maturity(stage_ready_to_burst)
+	return TRUE
+
+/obj/effect/alien/egg/gas
+	desc = "It looks like a weird egg"
+	name = "Hugger egg"
+	icon_state = "egg_hugger"
+	maturity_time = 15 SECONDS
+	stage_ready_to_burst = 1
```

Comment:
Right now gas eggs start out mature, since they already become mature while they're hugger eggs pre-gas. I'd say just set maturity_stage in the define but you're using advance to set the turf triggers. Maybe check whether mat_stage == burst_stage in init?